packages/suite/src/views/settings/SettingsGeneral/Labeling.tsx

```diff
@@ -39,7 +40,10 @@ export const Labeling = () => {
     // - Labeling enabled without any device connected
     // - Labeling enabled with the device connected inside Settings
     // The initialization of Labeling then start when user select a Wallet.
-    const isDisabled = (!!device && !metadata.enabled && isLocked()) || device?.mode !== 'normal';
+    const isDisabled =
+        isDiscoveryRunning ||
+        (!!device && !metadata.enabled && isLocked()) ||
+        device?.mode !== 'normal';
```

Comment:
On the device settings page, the `isLocked()` fn from the `useDevice()` hook is being used, so let's use it here as well for consistency? :) 

Also, maybe the whole condition can just be this? 

```
const isDisabled = isLocked() || device?.mode !== 'normal';
```

The `(!!device && !metadata.enabled && isLocked())` simply means `when a locked device with metadata disabled` is present – disable the Switch. But even if there's a locked device with metadata enabled is present it should get disabled 🤔

Reply:
> On the device settings page, the isLocked() fn from the useDevice() hook is being used, so let's use it here as well for consistency? :)

Looks like it's working. I'm not sure if we are not missing something, but let's be brave 😀

Reply:
* if there is no `device` then `isLocked() === false` by default
* metadata being enabled or disabled should not play a role if `isLocked() === true`
* when discovery is happening `isLocked() === true` 

So let's see ;DDD